### PR TITLE
[WIP] Endpoint name mandatory constructor parameter on EndpointConfiguration

### PR DESCRIPTION
**Pull request by andreasohlund**
Not flagged as v6 core yet so consider this a spike for now

Like discussed in https://github.com/Particular/NServiceBus/issues/2477.

### WIP

- [x] Add obsoletes
- [ ] Doco pull
- [x] [WIP pull on the windows service bootstrapper](https://github.com/Particular/NServiceBus.Bootstrap.WindowsService/pull/3)
- [x] [NSB Host WIP pull](https://github.com/Particular/NServiceBus.Host/pull/58)
- [x] [Azure Host WIP pull](https://github.com/Particular/NServiceBus.Host.AzureCloudService/pull/23)

Closes https://github.com/Particular/NServiceBus/issues/2477
Closes https://github.com/Particular/NServiceBus/issues/3549

### CODE REVIEW

**Comment by andreasohlund** [comment]
@Particular/nservicebus-maintainers can you do a preliminary review? (I'm adding the azure pull now and @SimonCropp will patch the doco tomorrow)

**Comment by andreasohlund** [comment]
@Particular/tf-v6-launch-plat-dev there seems to be consensus that this is a low hanging fruit, marking this as v6. I will go in tomorrow if it passes review and @SimonCropp doesn't hit any roadblocks for doco.

@SimonCropp I'll help with the upgrade guide, lets talk tomorrow

File: src/NServiceBus.Core/EndpointConfiguration.cs [hunk 1]
```diff
@@ -21,14 +20,19 @@ namespace NServiceBus
     /// <summary>
     /// Configuration used to create an endpoint instance.
     /// </summary>
-    public class EndpointConfiguration : ExposeSettings
+    public  class EndpointConfiguration : ExposeSettings
```
Comment (timbussmann) [inline src/NServiceBus.Core/EndpointConfiguration.cs]: this class has so much space it should be named BusinessClass 

sorry, that was so bad.

**Comment by timbussmann** [comment]
looks good to me at first glance. Love all the removed code :+1:

**Comment by andreasohlund** [comment]
WIP pulls for the hosts are up and linked

@SimonCropp should be good to go for a doco pull, assigned to you

File: src/NServiceBus.Core/EndpointConfiguration.cs (hunk 1 above)
Comment (andreasohlund) [inline src/NServiceBus.Core/EndpointConfiguration.cs]: fixed

**Comment by andreasohlund** [comment]
Rebased and squashed

File: src/NServiceBus.Core/obsoletes.cs [hunk 2]
```diff
@@ -457,7 +468,8 @@ public SendOptions ToSendOptions(string replyToAddress)
         [ObsoleteEx(
             Message = "Not used anymore",
             RemoveInVersion = "7.0",
-            TreatAsErrorFromVersion = "6.0")] public const string OriginalReplyToAddress = "NServiceBus.Timeout.ReplyToAddress";
+            TreatAsErrorFromVersion = "6.0")]
```
Comment (danielmarbach) [inline src/NServiceBus.Core/obsoletes.cs]: Did you do a reformat on this file? Cause these two changes are not relevant to this PR

**Comment by danielmarbach** [comment]
Like it. All good. The one comment I made is more curiosity

File: src/NServiceBus.Core/obsoletes.cs (hunk 2 above)
Comment (andreasohlund) [inline src/NServiceBus.Core/obsoletes.cs]: Yes, did a reformat without thinking about it, want me to revert that?

**Comment by andreasohlund** [comment]
I forgot the win service bootstrapper, will do a WIP pull there as well

**Comment by andreasohlund** [comment]
Bootstrapper pull is up https://github.com/Particular/NServiceBus.Bootstrap.WindowsService/pull/3

**Comment by timbussmann** [comment]
title still marks this as WIP?

**Comment by andreasohlund** [comment]
My bad, I forgot to remove the WIP. I told @SimonCropp to merge once he had the doco lined up since this had to be delicately timed with doco updates. This WIP was only there until simon had the doco ready. Does this make sense?

I should have mentioned the plans in the description of this pull :(

**Comment by danielmarbach** [comment]
I removed the [WIP]

**Pull request merged**